File: grind75/Jan13/11.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        # two pointers
        n = len(height)
        left, right = 0, n-1
        max_area = 0
        while left < right:
            if height[left] < height[right]:
                area = (right - left) * height[left]
                max_area = max(max_area, area)
                left += 1
            else:
                area = (right - left) * height[right]
                max_area = max(max_area, area)
                right -= 1
        return max_area

File: grind75/Jan13/11.go
func maxArea(height []int) int {
    i,j := 0, len(height)-1
    ret := 0
    for i < j {
        ret = max(ret, (j-i)*min(height[i], height[j]))
        if height[i] < height[j] {
            i++
        } else {
            j--
        }
    }
    return ret
}
